syntax = "proto3";

package exchange;

message NewOrder {
	enum SideType {
	  BUY = 0;
	  SELL = 1;
	}
	uint32 user = 1;
	string symbol = 2; 
	uint32 price = 3;
	uint32 quantity = 4;
	SideType side = 5;
	uint32 userOrder = 6;
}

message CancelOrder {
	uint32 user = 1;
	uint32 userOrder = 2;
}

message FlushOrder {
}

message ExchangeMessage {
    uint64 sequenceNumber = 1;
    oneof command {
    	NewOrder newOrder = 2;
    	CancelOrder cancelOrder = 3;
    	FlushOrder flushOrder = 4;
    }
}